File: java/low-level-design/src/com/educosys/parkinglot/Vehicle.java
package com.educosys.parkinglot;

public class Vehicle {
    int regNo;
    VehicleType type;

    Vehicle() {}

    Vehicle(int regNo, VehicleType type) {
        this.regNo = regNo;
        this.type = type;
    }

    public int getRegNo() {
        return regNo;
    }

    public void setRegNo(int regNo) {
        this.regNo = regNo;
    }

    public VehicleType getType() {
        return type;
    }

    public void setType(VehicleType type) {
        this.type = type;
    }
}

File: c++/parking-lot-lld/Vehicle.cpp
#include "Vehicle.h"
using namespace std;

Vehicle::Vehicle(string licencePlate, VehicleType type)
{
    this->licencePlate = licencePlate;
    this->type = type;
}

VehicleType Vehicle::getType(){
    return type;
}

string Vehicle::getLicencePlate(){
    return licencePlate;
}
